docs/v1-proposed-api.md

```diff
@@ -0,0 +1,145 @@
+# Proposed Changes for Hive V1 API
+
+## ClusterDeployment
+
+ 1. Require an InstallConfig ConfigMap, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
+   * GOAL: Allow quicker exposure to new Installer features without needing to mirror in Hive's API.
+   * GOAL: Remaining ClusterDeployment.Spec should be things that can be modified post-install.
+   * Should Hive parse the InstallConfig to obtain information it needs, eliminating the need for the caller to duplicate information in ClusterDeployment and InstallConfig?
+     * Or should Hive be explicitly given all the information it needs to adopt and maintain a cluster in it's ClusterDeployment, requiring the caller to duplicate a fair portion of InstallConfig in each.
```

Comment:
I would think we would want define the API but not require duplicating data. The user should be able to set commonly used values in the ClusterDeployment and leave them out of the InstallConfig, but have Hive add them to the install config it passes to the installer. That way we give the user the benefit of validation, a well-documented API, and some ease of use, but leave open the ability to extend Hive to support new platforms quickly (by passing a more complete InstallConfig) and to add seldom-used values to the config for special cases.

Reply:
Team discussion went the opposite direction, take a full InstallConfig, copyback portions to ClusterDeployment if they can be managed day 2 and beyond by Hive.